examples/dataflow-flex-python/README.md

```diff
@@ -0,0 +1,202 @@
+Example also contains a DAG which can be used to trigger Dataflow job from composer. It also demonstrates
+how we can use cloudbuild to implement CI/CD for this dataflow job.
+
+There are two docker files in the repo, the Dockerfile present at the root of directory is to create a image
+with pre-installed dependencies required for workers to run the job. This image is used by worker at job run time to
+avoid dependency downloading, its helpful in scenarios where workers dont have internet connectivity.
+
+Another Dockerfile in directory `df-sample-xml-to-bq` creates an image to be used by launcher in order to launch the
+dataflow job.
+
+
+### Setup
+```
+export PROJECT_ID=<project_id>
+export PROJECT_NUMBER=$(gcloud projects describe $PROJECT_ID --format="value(projectNumber)")
+export HOST_PROJECT_ID=<HOST_PROJECT_ID>
+export INPUT_BUCKET_NAME=pw-df-input-bkt
+export STAGING_BUCKET_NAME=pw-df-temp-bkt
+export LOCATION=us-central1
+export BQ_DATASET=bqtoxmldataset
+export NETWORK=shared-vpc
+export SUBNET=bh-subnet-usc1
+export REPO=dataflowf-image-repo
+export DF_WORKER_SA=dataflow-worker-sa@$PROJECT_ID.iam.gserviceaccount.com
+
+```
+
+#### Setup IAM
+```
+# Create service account for dataflow workers and launchers
+gcloud iam service-accounts create dataflow-worker-sa --project=$PROJECT_ID
+
+# Assign dataflow worker permissions
+gcloud projects add-iam-policy-binding $PROJECT_ID --member "serviceAccount:dataflow-worker-sa@$PROJECT_ID.iam.gserviceaccount.com" --role roles/dataflow.worker
+
+# Assign Object viewer permissions in order to read the data from cloud storage
+gcloud projects add-iam-policy-binding $PROJECT_ID --member "serviceAccount:dataflow-worker-sa@$PROJECT_ID.iam.gserviceaccount.com" --role roles/storage.objectViewer
+
+# Assign Object viewer permissions in order to Create temp files cloud storage
+gcloud projects add-iam-policy-binding $PROJECT_ID --member "serviceAccount:dataflow-worker-sa@$PROJECT_ID.iam.gserviceaccount.com" --role roles/storage.objectCreator
+
+# Assign Service Account User permissions
+gcloud projects add-iam-policy-binding $PROJECT_ID --member "serviceAccount:dataflow-worker-sa@$PROJECT_ID.iam.gserviceaccount.com" --role roles/iam.serviceAccountUser
+
+# Assign BigQuery job ser permissions
+gcloud projects add-iam-policy-binding $PROJECT_ID --member "serviceAccount:dataflow-worker-sa@$PROJECT_ID.iam.gserviceaccount.com" --role roles/bigquery.jobUser
+
+# Assign bigquery data editor permissions
+gcloud projects add-iam-policy-binding $PROJECT_ID --member "serviceAccount:dataflow-worker-sa@$PROJECT_ID.iam.gserviceaccount.com" --role roles/bigquery.dataEditor
+
+# Assign Artifactory Reader Permissions
+gcloud projects add-iam-policy-binding $PROJECT_ID --member "serviceAccount:dataflow-worker-sa@$PROJECT_ID.iam.gserviceaccount.com" --role roles/artifactregistry.reader
+
+# Assign network user permissions on Host project
```

Comment:
Those are needed only if you are using a shared-vpc. I would mention

Reply:
added a comment